src/SerializerGenerator.php

```diff
@@ -240,6 +240,11 @@ private function generateCodeForArray(
                 $innerCode = $this->generateCodeForClass($subType->getClassMetadata(), $apiVersion, $serializerGroups, $arrayPath.'['.$index.']', $modelPath.'['.$index.']', $stack);
                 break;
 
+            case $subType instanceof PropertyTypeUnknown;
+                if ($this->configuration->shouldAssignUnknownArrays()) {
```

Comment:
i think this condition needs to be part of the `case` criteria, so `case $subType instanceof PropertyTypeUnknown && $this->configuration->shouldAssignUnknownArrays():` - otherwise when the flag is false, we do not get the previous error handling.

Reply:
Fixed via 90786c09325b9f0daeb4fa29d9f79e0f31535a45